pkg/util/instancemetadata/dev.go

```diff
@@ -25,12 +27,19 @@ func NewDev(checkEnv bool) (InstanceMetadata, error) {
 	}
 
 	environment := azure.PublicCloud
+	aroEnvironment := azureclient.PublicCloud
+
 	if value, found := os.LookupEnv("AZURE_ENVIRONMENT"); found {
 		var err error
 		environment, err = azure.EnvironmentFromName(value)
```

Comment:
Did you looked into extending this object with our custom object? Something like
```
type AROEnvironment struct {
azure.Environment
ActualCloudName:          "AzureUSGovernment",
GenevaMonitoringEndpoint: "https://gcs.monitoring.core.usgovcloudapi.net/",
AppSuffix:                "aro.azure.us",
}
```
So we could have one environment type and just enriched values?

Reply:
+1 on this. We have too many things that we call "env" or "environment" already.

In the addition to the above suggestion: is it worth looking into replacing `azure.EnvironmentFromName(...)` calls with `azureclient.EnvironmentFromName(...)` across the codebase?

Reply:
@mjudeikis Yes, I did that originally. However, to consistently use our extension, it required touching a ton of files.

```
is it worth looking into replacing azure.EnvironmentFromName(...) calls with azureclient.EnvironmentFromName(...)
```

@m1kola I didn't think so, which is why I ultimately decided to leave them decoupled.

I can change it back, though, if you're fine with either:

1. Only using our extension where needed (might get confusing as to when we use which one)
2. Replacing all usages with our extension (lots of file changes)

Another consideration is that there are a few vendored functions that specifically consume an `azure.Environment`, so even with our extension we'd have to access the inner type at times... not a problem, but it also raises the question of why they need to be coupled.

Reply:
this is what it looks like to use the extended type everywhere:
https://github.com/Azure/ARO-RP/commit/8dbaa207a59cd813574356b5278dc89945395199

Reply:
I'm switching back to extending `azure.Environment` as agreed upon.